src/AnalysisServices/AnalysisServices.Dataplane/Models/AsAzureDataplaneClient.cs

```diff
@@ -0,0 +1,231 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Rest;
+
+namespace Microsoft.Azure.Commands.AnalysisServices.Dataplane.Models
+{
+    /// <summary>
+    /// Contains properties and logic for <see cref="AsAzureDataplaneClient"/>.
+    /// All permutations of CallGetAsync, and CallPostAsync are in the partial class in order to make this partial class more readable.
+    /// </summary>
+    /// <remarks>Most modifications will likely be in this part of the class.</remarks>
+    public partial class AsAzureDataplaneClient : ServiceClient<AsAzureDataplaneClient>, IAsAzureHttpClient
+    {
+        /// <summary>
+        /// The base Uri of the service.
+        /// </summary>
+        public Uri BaseUri { get; set; }
+
+        /// <summary>
+        /// Credentials needed for the client to connect to Azure.
+        /// </summary>
+        public ServiceClientCredentials Credentials { get; private set; }
+
+        private Func<HttpClient> HttpClientProvider { get; set; }
+
+        // This may cause problems, but is necessary for mulitple inheritance.
+        public new HttpClient HttpClient { get; set; }
+
+        public AsAzureDataplaneClient(Uri baseUri, ServiceClientCredentials credentials, Func<HttpClient> httpClientProvider, params DelegatingHandler[] handlers) : base(handlers)
+        {
+            this.BaseUri = baseUri ?? throw new ArgumentNullException(nameof(baseUri));
+            this.Credentials = credentials ?? throw new ArgumentNullException(nameof(credentials));
+            this.HttpClientProvider = httpClientProvider ?? throw new ArgumentNullException(nameof(httpClientProvider));
+            this.Credentials.InitializeServiceClient(this);
+            this.resetHttpClient();
+        }
+
+        public AsAzureDataplaneClient(Uri baseUri, ServiceClientCredentials credentials, Func<HttpClient> httpClientProvider, HttpClientHandler rootHandler, params DelegatingHandler[] handlers) : base(rootHandler, handlers)
+        {
```

Comment:
can we call the above constructor here instead of duplicating code?

Reply:
Unfortunately, due to the differing base() calls, these constructors can't be combined.

Reply:
One of the constructors was unneeded, so it was removed.